yb-voyager/src/migassessment/sizing.go

```diff
@@ -375,6 +389,125 @@ func pickBestRecommendation(recommendations map[int]IntermediateRecommendation)
+		return recAllSharded, reasoning
+	}
+	if rec2HasFailure && !rec1HasFailure {
+		reasoning := "Selected colocated+sharded strategy as all-sharded strategy could not support the requirements: " + recCombined.FailureReasoning
+		return recCombined, reasoning
+	}
+
+	// Both have empty failure reasoning, proceed with normal comparison logic
+	var selectedRec, notSelectedRec IntermediateRecommendation
+	var reasoning string
+
+	// Check if recommendations have different VCPUsPerInstance and apply resultant_cores logic
+	vcpuConditionMet := false
+	if recCombined.VCPUsPerInstance != recAllSharded.VCPUsPerInstance {
+		// Calculate resultant_cores = VCPUsPerInstance * NumNodes
+		rec1ResultantCores := float64(recCombined.VCPUsPerInstance) * recCombined.NumNodes
+		rec2ResultantCores := float64(recAllSharded.VCPUsPerInstance) * recAllSharded.NumNodes
+
+		if rec1ResultantCores == rec2ResultantCores {
+			// When resultant_cores are equal, choose the one with higher VCPUsPerInstance
+			if recCombined.VCPUsPerInstance > recAllSharded.VCPUsPerInstance {
+				selectedRec = recCombined
+				notSelectedRec = recAllSharded
+				reasoning = fmt.Sprintf("\nSelected colocated+sharded strategy with %d VCPUs per instance (%.1f resultant cores) over all-sharded strategy with %d VCPUs per instance (%.1f resultant cores). ",
+					recCombined.VCPUsPerInstance, rec1ResultantCores, recAllSharded.VCPUsPerInstance, rec2ResultantCores)
+			} else {
+				selectedRec = recAllSharded
+				notSelectedRec = recCombined
+				reasoning = fmt.Sprintf("\nSelected all-sharded strategy with %d VCPUs per instance (%.1f resultant cores) over colocated+sharded strategy with %d VCPUs per instance (%.1f resultant cores). ",
+					recAllSharded.VCPUsPerInstance, rec2ResultantCores, recCombined.VCPUsPerInstance, rec1ResultantCores)
+			}
+			vcpuConditionMet = true
+		} else if rec1ResultantCores < rec2ResultantCores {
+			selectedRec = recCombined
+			notSelectedRec = recAllSharded
+			reasoning = fmt.Sprintf("\nSelected colocated+sharded strategy with %d VCPUs per instance (%.1f resultant cores) over all-sharded strategy with %d VCPUs per instance (%.1f resultant cores). ",
+				recCombined.VCPUsPerInstance, rec1ResultantCores, recAllSharded.VCPUsPerInstance, rec2ResultantCores)
+			vcpuConditionMet = true
+		} else if rec2ResultantCores < rec1ResultantCores {
+			selectedRec = recAllSharded
+			notSelectedRec = recCombined
+			reasoning = fmt.Sprintf("\nSelected all-sharded strategy with %d VCPUs per instance (%.1f resultant cores) over colocated+sharded strategy with %d VCPUs per instance (%.1f resultant cores). ",
+				recAllSharded.VCPUsPerInstance, rec2ResultantCores, recCombined.VCPUsPerInstance, rec1ResultantCores)
+			vcpuConditionMet = true
+		}
+	}
+
+	// If VCPUsPerInstance condition was not met, use original node comparison logic
+	if !vcpuConditionMet {
```

Comment:
do we really need this condition variable? seems like a simple if-else would work, no?